examples/nas/oneshot/spos/search.py

```diff
@@ -1,17 +1,88 @@
+
+import nni
+from nn_meter import load_latency_predictor
+import nni.retiarii.nn.pytorch as nn
 import nni.retiarii.strategy as strategy
-from nni.retiarii import serialize
+from nni.retiarii.evaluator.functional import FunctionalEvaluator
+from nni.retiarii.utils import original_state_dict_hooks
+from nni.retiarii.oneshot.pytorch.utils import AverageMeterGroup
 from nni.retiarii.experiment.pytorch import RetiariiExeConfig, RetiariiExperiment
-from torchvision import transforms
-from torchvision.datasets import CIFAR10
-from nn_meter import load_latency_predictor
 
-from network import ShuffleNetV2OneShot
-from utils import get_archchoice_by_model
+from network import ShuffleNetV2OneShot, load_and_parse_state_dict
+from utils import CrossEntropyLabelSmooth, accuracy, ToBGRTensor, get_archchoice_by_model
+
+logger = logging.getLogger("nni.spos.search")
+
+
+def retrain_bn(model, criterion, max_iters, log_freq, loader):
+    with torch.no_grad():
+        logger.info("Clear BN statistics...")
+        for m in model.modules():
+            if isinstance(m, nn.BatchNorm2d):
+                m.running_mean = torch.zeros_like(m.running_mean)
+                m.running_var = torch.ones_like(m.running_var)
+
+        logger.info("Train BN with training set (BN sanitize)...")
+        model.train()
+        meters = AverageMeterGroup()
+        for step in range(max_iters):
+            inputs, targets = next(iter(loader))
+            inputs, targets = inputs.to('cuda'), targets.to('cuda')
+            logits = model(inputs)
+            loss = criterion(logits, targets)
+            metrics = accuracy(logits, targets)
+            metrics["loss"] = loss.item()
+            meters.update(metrics)
+            if step % log_freq == 0 or step + 1 == max_iters:
+                logger.info("Train Step [%d/%d] %s", step + 1, max_iters, meters)
+
+
+def test_acc(model, criterion, log_freq, loader):
+    logger.info("Start testing...")
+    model.eval()
+    meters = AverageMeterGroup()
+    start_time = time.time()
+    with torch.no_grad():
+        for step, (inputs, targets) in enumerate(loader):
+            inputs, targets = inputs.to('cuda'), targets.to('cuda')
+            logits = model(inputs)
+            loss = criterion(logits, targets)
+            metrics = accuracy(logits, targets)
+            metrics["loss"] = loss.item()
+            meters.update(metrics)
+            if step % log_freq == 0 or step + 1 == len(loader):
+                logger.info("Valid Step [%d/%d] time %.3fs acc1 %.4f acc5 %.4f loss %.4f",
+                            step + 1, len(loader), time.time() - start_time,
+                            meters.acc1.avg, meters.acc5.avg, meters.loss.avg)
+    return meters.acc1.avg
+
+
+def evaluate_acc(class_cls, criterion, args, train_dataset, val_dataset):
+    model = class_cls()
+    with original_state_dict_hooks(model):
+        model.load_state_dict(load_and_parse_state_dict(), strict=False)
```

Comment:
does user need special logic in `load_and_parse_state_dict` when loading a supernet checkpoint?

Reply:
Yes there should be a parameter for users to change the default location. Have fixed it.